src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -767,8 +768,8 @@ private void OnUpdateGraphCompleted(AsyncTask task)
             // Dispatch the failure message display for execution on UI thread.
             // 
             EvaluationCompletedEventArgs e = task.Exception == null || IsTestMode
-                ? new EvaluationCompletedEventArgs(true)
-                : new EvaluationCompletedEventArgs(true, task.Exception);
+                ? new EvaluationCompletedEventArgs(true,messages.Keys.Select(x=>x.ToString()),null)
+                : new EvaluationCompletedEventArgs(true, messages.Keys.Select(x => x.ToString()), task.Exception);
```

Comment:
I noticed the keys were converted into string and later converted back to Guids, any reason for not passing the keys as a collection?

Reply:
yeah, this was just lazy, will make this cleaner.

Reply:
So `messages` here are any messages related to warnings, errors or info that need to be displayed on the node?

Reply:
I think these are just runtime warnings and build warnings:
https://github.com/DynamoDS/Dynamo/blob/94326a0a020257303029bdfb25f0f79f9118b396/src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs#L712